feat(docs): update to docs from sphinx to starlight (#659)
algorandfoundation/puya: docs/src/content/docs/compiler.md

---
title: Compiler
description: PuyaPy compiler usage, options, and output formats
---

> The PuyaPy compiler is a multi-stage, optimising compiler that takes Algorand Python and prepares it for execution on the AVM. PuyaPy ensures the resulting AVM bytecode has execution semantics that match the given Python code. PuyaPy produces output that is directly compatible with [AlgoKit typed clients](https://github.com/algorandfoundation/algokit-cli/blob/main/docs/features/generate.md#1-typed-clients) to make deployment and calling easy (among other formats).

Reviewer: Same as before. Linking a repo is not a good idea.

Reviewer: ```suggestion
The PuyaPy compiler is a multi-stage, optimising compiler that takes Algorand Python and prepares it for execution on the AVM. PuyaPy ensures the resulting AVM bytecode has execution semantics that match the given Python code. PuyaPy produces output that is directly compatible with [AlgoKit typed clients](https://algorandfoundation.github.io/algokit-cli/features/generate/#1-typed-clients) to make deployment and calling easy (among other formats).
```

### Defining template values

> [Template Variables](/puya/api/algopy/algopy/#class-templatevar), can be replaced with literal values during compilation to bytecode using the `--template-var` option.

Reviewer: Wrong link

Reviewer: /puya/api/algopy/algopy/#templatevar--_templatevargeneric

Reviewer: ```suggestion
[Template Variables](/puya/api/algopy/algopy/#templatevar--_templatevargeneric), can be replaced with literal values during compilation to bytecode using the `--template-var` option.
```

> Additionally, Algorand Python functions that create AVM bytecode, such as [compile_contract](/puya/api/algopy/algopy/) and [compile_logicsig](/puya/api/algopy/algopy/), can also provide the specified values.

Reviewer: `/puya/api/algopy/algopy/#compile_contract` and `/puya/api/algopy/algopy/#compile_logicsig`

Reviewer: ```suggestion
Additionally, Algorand Python functions that create AVM bytecode, such as [compile_contract](/puya/api/algopy/algopy/#compile_contract) and [compile_logicsig](/puya/api/algopy/algopy/#compile_logicsig), can also provide the specified values.
```
